File: src/main/kotlin/com/sastudios/gatekeeper/websocket/WebSocketConfig.java
package com.sastudios.gatekeeper.websocket;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketConfig implements WebSocketConfigurer {
    @Autowired
    DocumentWebSocketHandler handler;
    public void registerWebSocketHandlers(WebSocketHandlerRegistry r) {
        r.addHandler(handler, "/ws/documents").setAllowedOrigins("*");
    }
}

File: src/main/kotlin/com/sastudios/gatekeeper/config/WebSocketConfig.kt
package com.sastudios.gatekeeper.config

import com.sastudios.gatekeeper.websocket.DocumentWebSocketHandler
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry


@Configuration
@EnableWebSocket
class WebSocketConfig(
    private val documentWebSocketHandler: DocumentWebSocketHandler
) : WebSocketConfigurer {

   override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(documentWebSocketHandler, "/ws/doc/{docId}")
            .setAllowedOrigins("*")
    }
}
